adding EFS_ENCRYPTED_CHECK rule

rafihm opened the pull request:
I confirm these files are made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)

*Issue #, if available:*
#129 
*Description of changes:*

jongogogo commented on python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py:
```diff
@@ -0,0 +1,448 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Not the latest RDK template

jongogogo commented on python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py:
```diff
@@ -0,0 +1,448 @@
+
+    # If parameter is given check whether it's ARN, else ignore it as it is optional paramter.
+    if 'KmsKeyId' not in rule_parameters:
+        return False
```
I prefer to keep the output of functions constant. Always return a dict in this case.

Start with 
```python
if 'KmsKeyId' not in rule_parameters:
        return {}

jongogogo commented on python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py:
```diff
@@ -0,0 +1,448 @@
+    # If parameter is given check whether it's ARN, else ignore it as it is optional paramter.
+    if 'KmsKeyId' not in rule_parameters:
+        return False
+    if rule_parameters['KmsKeyId'] and 'arn:aws:kms' not in rule_parameters['KmsKeyId']:
```
The `rule_parameters['KmsKeyId']` is always True due to your guard clause above. You can remove it

jongogogo commented on python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py:
```diff
@@ -0,0 +1,448 @@
+        return False
+    if rule_parameters['KmsKeyId'] and 'arn:aws:kms' not in rule_parameters['KmsKeyId']:
+        raise ValueError('Invalid value for paramter KmsKeyId, Expected KMS Key ARN')
+    if 'KmsKeyId' in rule_parameters and "arn:aws:kms" in rule_parameters['KmsKeyId']:
```
`'KmsKeyId' in rule_parameters` is always True
`"arn:aws:kms" in rule_parameters['KmsKeyId']` is always True

You can just `return rule_parameters` without a `if` statement

jongogogo commented on python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK_test.py:
```diff
@@ -0,0 +1,253 @@
+import sys
```
Not the latest RDK template

jongogogo commented on python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK_test.py:
```diff
@@ -0,0 +1,253 @@
+    rule_invalid_value_parameter = '{"KmsKeyId": "asdfa97asf8a0sf09sa8df0a98sd0f8as0f8d0"}'
+    rule_empty_parameter = '{"KmsKeyId": ""}'
+
+    invoking_event_iam_role_sample = '{"configurationItem":{"relatedEvents":[],"relationships":[],"configuration":{},"tags":{},"configurationItemCaptureTime":"2018-07-02T03:37:52.418Z","awsAccountId":"123456789012","configurationItemStatus":"ResourceDiscovered","resourceType":"AWS::IAM::Role","resourceId":"some-resource-id","resourceName":"some-resource-name","ARN":"some-arn"},"notificationCreati ...
```
Delete what is not necessary.

rafihm commented:
Change Log 1:
 - Changed the template to latest version
 - Modified gherkin parameter section to remove the table format
 - Modified gherkin scenario 2
 - returning None if no file system exists
 - correcting type in annotation
 - removed maxItems from API call
 - removed extra variables present in the test file

Let me know if any modification requires. Thanks

jongogogo commented on python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py:
```diff
@@ -0,0 +1,450 @@
+    for each_efs in all_file_systems:
+        # check if file system is encrypted
+        if not each_efs['Encrypted']:
+            evaluations.append(build_evaluation(each_efs['FileSystemId'], 'NON_COMPLIANT', event, annotation='EFS is not encryprted'))
```
Typo in the annotation. Please call the services by the full name. Add a final dot.

jongogogo commented on python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py:
```diff
@@ -0,0 +1,450 @@
+        if each_efs['KmsKeyId'] == valid_rule_parameters:
+            evaluations.append(build_evaluation(each_efs['FileSystemId'], 'COMPLIANT', event))
+        else:
+            evaluations.append(build_evaluation(each_efs['FileSystemId'], 'NON_COMPLIANT', event, annotation='KMS key used to encrypt the EFS is not mathcing the KMS Key provided in the parameter'))
```
Same, find a better wordings.

rafihm commented:
Modified Rule parameter and annotations wordings.
